core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -178,18 +178,19 @@ private FileSystemContext(Subject subject) {
    * Initializes the context. Only called in the factory methods and reset.
    *
    * @param masterInquireClient the client to use for determining the master
-   * @param enableUserMetricsCollection if enables the user metrics collection
+   * @param configuration the instance configuration
    */
   private synchronized void init(MasterInquireClient masterInquireClient,
-      boolean enableUserMetricsCollection) {
+      InstancedConfiguration configuration) {
     mMasterInquireClient = masterInquireClient;
     mFileSystemMasterClientPool =
         new FileSystemMasterClientPool(mParentSubject, mMasterInquireClient);
     mBlockMasterClientPool = new BlockMasterClientPool(mParentSubject, mMasterInquireClient);
     mClosed.set(false);
 
     // Only send metrics if enabled and the port is set (can be zero when tests are setting up).
-    if (enableUserMetricsCollection && Configuration.getInt(PropertyKey.MASTER_RPC_PORT) != 0) {
+    if (configuration.getBoolean(PropertyKey.USER_METRICS_COLLECTION_ENABLED)
+        && Configuration.getInt(PropertyKey.MASTER_RPC_PORT) != 0) {
```

Comment:
can we remove the `Configuration.getInt(PropertyKey.MASTER_RPC_PORT) != 0` now?

Reply:
yes, good call